src/components/ebay-calendar/component.js

```diff
@@ -0,0 +1,316 @@
+    }
+
+    getLastVisibleISO(input = this.input) {
+        const baseDate = fromISO(this.state.baseISO);
+        return toISO(
+            new Date(
+                baseDate.getFullYear(),
+                baseDate.getMonth() + this.state.offset + (input.numMonths || 1),
+                0
+            )
+        );
+    }
+
+    getFirstActiveISO(input = this.input) {
+        let iso = this.getFirstVisibleISO();
+        const lastVisible = this.getLastVisibleISO(input);
+        while (iso <= lastVisible && this.isDisabled(iso)) {
+            iso = offsetISO(iso, 1);
+        }
+        return iso;
+    }
+
+    getLastActiveISO(input = this.input) {
+        let iso = this.getLastVisibleISO(input);
+        const firstVisible = this.getFirstVisibleISO();
+        while (iso >= firstVisible && this.isDisabled(iso)) {
+            iso = offsetISO(iso, -1);
+        }
+        return iso;
+    }
+
+    /**
+     * @param {Date} date
+     */
+    monthTitle(date) {
+        const formatter = new Intl.DateTimeFormat(localeOverride(this.input.locale), {
+            month: 'long',
+            year: 'numeric',
+        });
+        return formatter.format(date);
+    }
+
+    /**
+     * @param {boolean | void} focus
+     */
+    prevMonth(focus) {
+        if (this.state.disableBefore && this.getFirstVisibleISO() <= this.state.disableBefore) {
+            return false;
+        }
+
+        this.state.offset--;
+        let newTabindexISO = this.state.tabindexISO;
+        const lastActiveISO = this.getLastActiveISO();
+        if (this.state.tabindexISO > lastActiveISO) {
+            newTabindexISO = this.state.tabindexISO = lastActiveISO;
+        }
+        if (focus) {
+            this.setTabindexAndFocus(newTabindexISO);
+        }
+
+        this.emit('month', { iso: toISO(this.getMonthDate(this.state.offset)) });
+
+        return true;
+    }
+
+    /**
+     * @param {boolean | void} focus
+     */
+    nextMonth(focus) {
+        if (this.state.disableAfter && this.getLastVisibleISO() >= this.state.disableAfter) {
+            return false;
+        }
+
+        this.state.offset++;
+        let newTabindexISO = this.state.tabindexISO;
+        const firstActiveISO = this.getFirstActiveISO();
+        if (this.state.tabindexISO < firstActiveISO) {
+            newTabindexISO = this.state.tabindexISO = firstActiveISO;
+        }
+        if (focus) {
+            this.setTabindexAndFocus(newTabindexISO);
+        }
+
+        this.emit('month', {
+            iso: toISO(this.getMonthDate(this.state.offset + (this.input.numMonths || 1))),
+        });
+
+        return true;
+    }
+
+    /**
+     * @param {DayISO} iso
+     */
+    setTabindexAndFocus(iso) {
+        this.state.tabindexISO = iso;
+        // After UI updates, focus on the new tabindex date
+        setTimeout(() => {
+            /** @type {HTMLButtonElement | undefined} */ (this.getEl(iso))?.focus();
```

Comment:
❗ This appears to be called `onKeyDown`, which means every keydown (with conditionals met) we are traversing the DOM and selecting the element - very costly. We should cache the element on instantiation of the component and simply reference it where needed. This should be done for other elements as well.

Reply:
This is not looking for a single element... Each time a key is pressed and conditions are met, we need to focus on a _new_ day. Unless I'm misunderstanding, we would only benefit from caching here if the user clicked the left/right or up/down arrows repeatedly to continuously return to the same element.

Reply:
Oh wow! Ok, but not great. Wasn't there a native keyboard solution for that we were considering during the markup determination phase?

Reply:
I think that was going to be done later.